tests/Offer/CommandHandlers/UpdateAvailableFromHandlerTest.php

```diff
@@ -103,4 +108,45 @@ public function updateAvailableFromDataProvider(): array
             ],
         ];
     }
+
+    /**
+     * @test
+     */
+    public function it_modifies_available_from_in_the_past_to_now(): void
+    {
+        if (!$this->traceableEventStore instanceof TraceableEventStore) {
+            return;
+        }
+
+        $eventId = '300ed979-03d8-48a0-859a-d03aeab0fa6a';
+
+        $eventCreated = new EventCreated(
+            $eventId,
+            new Language('nl'),
+            new Title('Permanent Event'),
+            new EventType('0.50.4.0.0', 'concert'),
+            new LocationId('d0cd4e9d-3cf1-4324-9835-2bfba63ac015'),
+            new Calendar(CalendarType::PERMANENT())
+        );
+
+        $updateAvailableFrom = new UpdateAvailableFrom($eventId, new DateTimeImmutable('2010-11-15T11:22:33+00:00'));
+
+        $this->scenario
+            ->withAggregateId($eventId)
+            ->given([$eventCreated])
+            ->when($updateAvailableFrom);
+
+        $events = $this->traceableEventStore->getEvents();
+
+        $this->assertCount(1, $events);
+        $this->assertInstanceOf(AvailableFromUpdated::class, $events[0]);
+
+        /** @var AvailableFromUpdated $availableFromUpdated */
+        $availableFromUpdated = $events[0];
+        $this->assertEquals($eventId, $availableFromUpdated->getItemId());
+
+        $nowTimestamp = (new DateTimeImmutable())->getTimestamp();
+        $availableFromTimeStamp = $availableFromUpdated->getAvailableFrom()->getTimestamp();
+        $this->assertTrue($nowTimestamp - $availableFromTimeStamp < 1);
```

Comment:
You could use the `->diff()` method on `DateTimeImmutable` to get a `DateInterval` object to do the assertions, but I guess this works as well 👍

Reply:
True, but then I still had to interpret the `DateInterval` and convert that to a total amount in seconds.